pkg/problemdaemon/disk.go

```diff
@@ -0,0 +1,164 @@
+/*
```

Comment:
I initially thought that you'll have another level of plugin mechanism for compile time plug-in metrics, but it seems that you decide to reuse the problem daemon plugin mechanism.

If this is the case, can:
1) Move the logic into its own package (assuming Register is public);
2) Consider whether we just want this to be a "disk monitor" or "system stats monitor". Let's say we need some memory stats in the future, do we want to add another problem daemon?

Reply:
> Do you plan to provide a compile time mechanism to exclude problem daemons that are not needed for some users?

Yes. I have two ideas to achieve that, but not decided yet. I plan to submit a following PR to address the "Implement option for disabling problem daemons at compile time" action item at #284 

I don't think it's very urgent yet. So planning to do it in a follow up PR.

> Consider whether we just want this to be a "disk monitor" or "system stats monitor". 

> A monolithic "system stats monitor" cannot satisfy such requirement.

I plan to introduce a config file to select the collected metrics anyway. So we can achieve the granularity through the config file.

I'm calling the problem daemon "system stats monitor" for now. We can allow it to evolve over time.